### PR TITLE
Unit test framework

### PR DESCRIPTION
**Pull request by SpyTec**
Root `tests/unit` folder for tests. Tests themselves need to import the file and trigger an event to call the function

Part of #104

### CODE REVIEW

File: tests/unit/modules/support.test.js [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+        expect(window.TB.Module.mock.results[0].value.settings.enabled.default).toBeTruthy();
+    });
+    it('sets name to enabled.hidden to true', () => {
+        const event = new CustomEvent('TBModuleLoaded');
+        window.dispatchEvent(event);
+        expect(window.TB.Module.mock.results[0].value.settings.enabled.hidden).toBeTruthy();
+    });
+    describe('init()', () => {
+        beforeEach(() => {
+            window.TBCore.debugInformation.mockClear;
```
Comment (k3n) [inline tests/unit/modules/support.test.js]: Is this supposed to be a method call?

File: tests/unit/modules/support.test.js [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+        });
+        it('calls TBCore.debugInformation()', () => {
+            const event = new CustomEvent('TBModuleLoaded');
+            window.dispatchEvent(event);
+            window.TB.Module.mock.results[0].value.init();
+            expect(window.TBCore.debugInformation).toHaveBeenCalled();
+        });
+        it('calls updates textArea with debug info on r/toolbox link submit', () => {
+            window.history.pushState({}, 'Page Title', '/r/toolbox/submit');
```
Comment (k3n) [inline tests/unit/modules/support.test.js]: It looks like this isn't getting cleaned up after the test -- `window.history.length === 1` prior to this line, and `2` afterwards... and if you add another test it's still `2`.

File: tests/unit/modules/support.test.js (hunk 1 above)
Comment (creesch) [reply tests/unit/modules/support.test.js]: Sonarcloud [agrees](https://sonarcloud.io/project/issues?id=toolbox-team_reddit-moderator-toolbox&open=AW3wZwzm1OSLGlZ3SMXq&pullRequest=185&resolved=false&types=BUG) with @k3n here ;)

**Comment by creesch** [comment]
Had a quick lookover and looks good to me.

Probably a discussion for something outside of this PR but I am wondering if we should start building tests for modules or maybe first for the utility classes and objects?

File: tests/unit/modules/support.test.js (hunk 1 above)
Comment (SpyTec) [reply tests/unit/modules/support.test.js]: Yes!

File: tests/unit/modules/support.test.js (hunk 2 above)
Comment (SpyTec) [reply tests/unit/modules/support.test.js]: I think we should be using `replaceState` instead to avoid this

**Comment by SpyTec** [comment]
I'm gonna see if we can do this in the env file to avoid having to reset mocks ourselves

```js
beforeEach(() => {
    jest.clearAllMocks();
});
```

> Probably a discussion for something outside of this PR but I am wondering if we should start building tests for modules or maybe first for the utility classes and objects?

Answered in #104

**Pull request merged**